### PR TITLE
Add Concourse PR pipeline

### PR DESCRIPTION
**Pull request by karlbaker02**
This PR adds CI to run tests when PRs are raised, using Concourse and the GitHub PR Resource (https://github.com/telia-oss/github-pr-resource).

It also refactors `EdgeWeightExtractor` to only initialise the BigQuery client when creating a new instance of this class.

### CODE REVIEW

File: concourse.yml [hunk 1]
```diff
@@ -9,22 +9,81 @@ groups:
...
+              - -c
+              - |
+                set -e pipefail
+
+                cd related-links-pr
+
+                python -V
+                apk update
+                apk add --no-cache bzip2-dev coreutils dpkg-dev dpkg expat-dev findutils gcc gdbm-dev libc-dev libffi-dev libnsl-dev libtirpc-dev linux-headers make ncurses-dev openssl-dev pax-utils readline-dev sqlite-dev tcl-dev tk tk-dev util-linux-dev xz-dev zlib-dev
+                apk add pyenv
```
Comment (issyl0) [inline concourse.yml]: Can you find/make a Docker image that has these already installed? That would seem cleaner.

File: concourse.yml [hunk 2]
```diff
@@ -9,22 +9,81 @@ groups:
...
+        params:
+          path: related-links-pr
+          status: pending
+      - task: run-tests
+        config:
+          platform: linux
+          image_resource:
+            type: docker-image
+            source:
+              repository: govsvc/task-toolbox
```
Comment (issyl0) [inline concourse.yml]: Interesting choice...

File: concourse.yml [hunk 3]
```diff
@@ -9,22 +9,81 @@ groups:
...
+          inputs:
+            - name: related-links-pr
+          run:
+            path: bash
+            args:
+              - -c
+              - |
+                set -e pipefail
+
+                cd related-links-pr
```
Comment (issyl0) [inline concourse.yml]: You can set `dir:` to the directory you want to run things in in the inputs instead of `cd`ing, I think.

**Comment by karlbaker02** [comment]
@issyl0 Thanks for your comments. This is a WIP PR (I should have made this draft) - made it a PR so that I can quickly test against Concourse as a PR 🙂

**Comment by karlbaker02** [comment]
Thanks @barrucadu. `dir` is useful and cleans things up a bit, it feels like it's perhaps more of a personal preference thing - if you need to `cd` into many places in your Bash script when perhaps it might confuse more than it helps

**Pull request merged**